code/modules/projectiles/updated_projectiles/projectile.dm

```diff
@@ -533,11 +537,14 @@ Normal range for a defender's bullet resist should be something around 30-50. ~N
 	//Run armor check. We won't bother if there is no damage being done.
 	if( damage > 0 && !(P.ammo.flags_ammo_behavior & AMMO_IGNORE_ARMOR) )
 		var/armor //Damage types don't correspond to armor types. We are thus merging them.
-		switch(P.ammo.damage_type)
-			if(BRUTE) armor = P.ammo.flags_ammo_behavior & AMMO_ROCKET ? getarmor_organ(organ, "bomb") : getarmor_organ(organ, "bullet")
-			if(BURN) armor = P.ammo.flags_ammo_behavior & AMMO_ENERGY ? getarmor_organ(organ, "energy") : getarmor_organ(organ, "laser")
-			if(TOX, OXY, CLONE) armor = getarmor_organ(organ, "bio")
-			else armor = getarmor_organ(organ, "energy") //Won't be used, but just in case.
+		if(P.armor_type) //If the projectile is meant to test a specific armor type, we use that.
```

Comment:
Stick to one logic. If you want to revamp the whole system, be my guest, but having two parallel systems is pure code debt logic.

Reply:
6:17 AM] Surrealistik: one is default behaviour
[6:17 AM] Surrealistik: the other is an override
[6:17 AM] Surrealistik: it's not 'two parallel systems'

Reply:
this is dumb. remove it.

Reply:
Revamping the whole system is insanely out of scope